### ISSUE TITLE
CreateMissingTypeMaps with AddProfile

### ISSUE THREAD

**Issue by renanelias** [AutoMapper/AutoMapper#1228]
Hello,

I'm using the AutoMapper with Profiles and it is working very well. Recently, I found the CreateMissingTypeMaps configuration and, if it is working like I understand, will make my development faster, one time will not be necessary create simple mappings.

I have added to my configuration a Profile with AddProfile and, after that, I did the CreateMissingTypeMaps configuration.

Ex:
------- Creating configuration ------------
 var config = new MapperConfiguration(cfg =>
            {
               cfg.AddProfile<PersonServiceMapperProfile>();
                cfg.CreateMissingTypeMaps = true;
            });

------- Inside my PersonServiceMapperProfile ------
     CreateMap<PersonData, PersonViewModel>().ConvertUsing<PersonDataToProfileViewModel>();

When the CreateMissingTypeMaps configuration is set to true, the Profile not works anymore. My custom mapping is not called.

Sorry my english errors. How to solve this question?

Thanks in advance.
Renan

**Comment by TylerCarlson1** [comment]
In your profile set `profile.CreateMissingTypeMaps = true`  cfg is default profile and `cfg.AddProfile()` is second in list.  First one will make any map and second will never get called.

**Comment by renanelias** [comment]
I need the automapper work with my profiles and CreateMissingTypeMaps configuration set to true.

For some reason, when the cfg.CreateMissingTypeMaps = true, my custom mapping is not called! My profile is called and the configuration for my custom mapper is done, but is never called.

Have no difference with the CreateMissingTypeMaps set into the profile or into the root cfg.

**Comment by TylerCarlson1** [comment]
Can you provide an example again?  use ``` in each line before and after your code so it shows up as code.  Otherwise <> don't show up.  You can see if you did it right in preview.
```
public void Example()
{
   // code
}
```

**Comment by jbogard** [comment]
" if it is working like I understand, will make my development faster, one time will not be necessary create simple mappings."

That is not the purpose of CreateMissingTypeMaps, fyi. The purpose is when you don't know the types at compile time.

If you misspell anything, or remove anything, or modify anything, your mappings will silently skip those properties and you won't know this unless you're looking at every page and every field. You're supposed to call AssertConfigurationIsValid in a unit test to prevent silent errors.

In short, don't do this as a shortcut. It's a very bad idea. Just call CreateMap. It's one line of code, takes maybe 5 seconds tops to write. But the result is *so* much more obvious, explicit, and safe.

**Comment by renanelias** [comment]
OK jbogard! Understood!

Thank to you two by the answers.

I will call CreateMap for all my mappings.

**Comment by gtaylor44** [comment]
I have a mapping which is reused amongst other configurations. When I use AssertConfigurationIsValid on these other configurations, I get a mapping error. Setting CreateMissingTypeMaps resolves this issue and the unit test passes but as you mention, it can introduce silent bugs. 

What do you recommend I do here? What's the best way to deal with reusable configurations?

**Comment by jbogard** [comment]
Can you open a new issue? Then a few more details too.
